honeybee_grasshopper_energy/src/HB Model to OSM.py

```diff
@@ -155,26 +155,38 @@ def orphaned_warning(object_type):
     with open(sim_par_json, 'w') as fp:
         json.dump(sim_par_dict, fp)
 
+    # process any measures input to the component
+    measures = None if len(measures_) == 0 or measures_[0] is None else measures_
+    no_report_meas = True if measures is None else \
+        all(meas.type != 'ReportingMeasure' for meas in measures)
+
     # collect the two jsons for output and write out the osw file
     jsons = [model_json, sim_par_json]
-    osw = to_openstudio_osw(directory, model_json, sim_par_json, epw_file=_epw_file)
+    osw = to_openstudio_osw(directory, model_json, sim_par_json,
+                            additional_measures=measures, epw_file=_epw_file)
 
     # run the measure to translate the model JSON to an openstudio measure
-    if run_ > 0:
+    if run_ > 0 and not no_report_meas:  # everything must run with OS CLI
+        osm, idf = run_osw(osw, measures_only=False)
+        sql, zsz, rdd, html, err = output_energyplus_files(os.path.dirname(idf))
+    elif run_ > 0:  # no reporting measure; simulate separately from measure application
```

Comment:
What happens if there is a reporting measure? We run it afterwards? This is a question to educate myself about the difference.

Reply:
As you see in the comment there - if there is a reporting measure , everything must be run with the OpenStudio CLI, including the simulation.

Normally, when there's no reporting measure, we first call the OpenStudio CLI to run the measures that build the OpenStudio model from the JSON and edit the OSM and/or IDF. Then, we take the IDF that is output by this process and run it through EnergyPlus using the EnergyPlus executable instead of the OpenStudio CLI. This yields a few advantages:
1. We can add additional strings into the IDF before we simulate it, which we know is really important to our advanced users
2. The user can see the progress of the EnergyPlus simulation if we run it using the EnergyPlus executable, whereas the OpenStudio CLI is a black box and you never really know if it's simulating or how much longer the simulation is going to take.
3. If all that the user wants to do is export an OSM or IDF and not run the simulation, this setup easily accommodates this.

However, the only way to run a reporting measure is if the simulation is executed with OpenStudio CLI. So we are forced to abandon the benefits above in order to be able to run the reporting measure. I will ask Dan about this to confirm the next time that we talk but there's nothing in the OpenStudio CLI documentation or the Measure writers guide about how to run a reporting measure without running the simulation.  In fact, the way that many of the reporting measures are set up, they involve both code to edit the model before simulation and code to generate the report afterwards.  So these reporting measures definitely can't be run in the absence of a simulation.